### ISSUE TITLE
Assertion failed: "descendants exist with accessible content, innerContent cannot be used"

### ISSUE THREAD

**Issue by zepumph** [phetsims/ratio-and-proportion#444]
I hit this while working on https://github.com/phetsims/joist/issues/762. When changing the name of the discover screen dynamically, it sets the inner content on the screen button, which has two RatioHandNodes as icon in the pdom as `input` tags but `hidden`. Instead we need to remove it from the PDOM completely.

**Comment by zepumph** [comment]
One thought that we decided not to go for:
>@jessegreenberg and I discussed this, and we originally felt like this is assertion is there to prevent potential use that JG and I didn't expect and desire in the PDOM. Since we created a ParallelDOM option innerContent that combined textContent and innerHTML, the possibility for misuse is higher due to the added complexity. Thus, basically we decide the rules, and the most straight forward decision we thought of is the allow pdomVisible nested content in a node that uses innerContent.

This, we found, doesn't work! Because of assumptions in PDOMUtils.setTextContent, it could blow away the hidden children. We decided it wasn't worth breaking the assumption of no children + innerContent that we have built up over the years. 

After fuzzing with a new assertion to catch this when children are added to a node that already has inner content, we found three cases to handle:

1. In AboutDialog, PDOM paragraphs had children that were readingblocks, it turns out that we can just use the reading blocks as the pure description too. Commit coming soon.
2. In RAP, we have the hand nodes as sliders used for a screen icon (thus inside the nav bar screen `button` tag. 
3.  the same thing but for amplitude sliders in the discrete screen of fourier making waves.

For (2) and (3), we need to remove the icons completely from the PDOM, and not just hide them with pdomVisible, thus, setting `tagName: null` is the appropriate path forward. This also meant opening up the AccessibleValueHandler api to allow tagName: null as a provided option, only when removing it from the PDOM. sounds good to me!
